lib/src/async/future_stream.dart

```diff
@@ -47,7 +47,7 @@ class FutureStream<T> extends Stream<T> {
         _controller.close();
       }
       _controller = null;
-    });
+    }) as Future<Stream<T>>;
```

Comment:
Curious, why is `catchError` not a generic method? Seems like it could simply be:

```dart
Future/*<S>*/ catchError/*<S>*/(/*=S*/ onError(dynamic error, [dynamic stackTrace]),
                  {bool test(Object error)}) => null;
```

@leafpetersen @vsmenon